### PR TITLE
VM Matrix for Parallelization, moved CG support to job analyze

### PR DESCRIPTION
**Pull request by KarishmaGhiya**
## Description
- Added strategy matrix for VM Parallelization
- Moved CG support to job analyze
---
This checklist is used to make sure that common guidelines for a pull request are followed.
[x] Please add REST spec PR link to the SDK PR
[x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
[x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**
### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
[x] Title of the pull request is clear and informative.
[x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
[x] Pull request includes test coverage for the included changes.
### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
[x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
[x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
[x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by KarishmaGhiya** [comment]
@Azure/azure-sdk-eng for review

**Comment by scbedd** [comment]
@Azure/azure-sdk-eng

File: .azure-pipelines/client.yml
Comment (mikeharder) [inline .azure-pipelines/client.yml]: Please revert whitespace / newline changes.

File: .azure-pipelines/client.yml
Comment (mikeharder) [inline .azure-pipelines/client.yml]: Rename to `DotNetCoreVersion`

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: .azure-pipelines/client.yml
Comment (mikeharder) [inline .azure-pipelines/client.yml]: Remove `os.name` variable since it appears to be unused.

File: .azure-pipelines/client.yml
Comment (mikeharder) [inline .azure-pipelines/client.yml]: Rename to `VmImage` to match new style guideline.

**Review by mikeharder — APPROVED** [review]
Approved pending changes to variables

File: .azure-pipelines/client.yml
Comment (KarishmaGhiya) [reply .azure-pipelines/client.yml]: Actually we've been using similar namings in strategy matrix, in all other repos
https://github.com/Azure/azure-sdk-for-js/blob/master/.azure-pipelines/client.yml
Will we need to update others too then?

Comment (KarishmaGhiya) [reply .azure-pipelines/client.yml]: I'll change it to OSVmImage, for consistency with other repos. Because we'll make similar changesin other repos

Comment (mikeharder) [reply .azure-pipelines/client.yml]: I prefer `VmImage` to `OSVmImage`, but I'm fine with either to prevent blocking this PR for longer.  In addition to standardizing yaml style, we'll also want to standardize names for common variables.

File: .azure-pipelines/client.yml
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Can you please also add MacOs?

File: .azure-pipelines/client.yml
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Could you please rename this to `Build`?

**Review by mikeharder — APPROVED** [review]

File: .azure-pipelines/client.yml
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Is this space what broke the mac build?

**Review by weshaggard — APPROVED** [review]

File: .azure-pipelines/client.yml
Comment (mikeharder) [reply .azure-pipelines/client.yml]: We thought it might be, but it doesn't appear so.  We now suspect DevOps is currently having issues running jobs on Mac agents.

**Comment by weshaggard** [comment]
@KarishmaGhiya can you please exclude the macOS for now and get this PR merged so that we can unblock @AlexGhiondea to enable his live tests with PR https://github.com/Azure/azure-sdk-for-net/pull/5239?

**Comment by mikeharder** [comment]
@weshaggard: Let's try one more build to see if Mac is fixed.  Started here: https://dev.azure.com/azure-sdk/public/_build/results?buildId=4725

**Comment by mikeharder** [comment]
Mac build appears to be working.

**Pull request merged**
